include/agent.h

```diff
@@ -0,0 +1,96 @@
+#ifndef _ABM_AGENT_H_
+#define _ABM_AGENT_H_
+
+#include <memory>
+#include <vector>
+
+#include "config.h"
+#include "graph.h"
+#include "agent.h"
+
+namespace abm {
+
+//! Agent class that has id, origin, destination, departure time, current node and compute path from current node to destination.
+class Agent {
+ public:
+  //! Construct an agent class
+  //! \param[in] graph Pass a reference of the graph that the agent does routing on
+  explicit Agent(const std::shared_ptr<abm::Graph>& graph) : graph_{graph} {};
+
+  //! Set id 
+  //! \param[in] agent_id Agent id assigned to the agent object
+  void set_id(graph::vertex_t agent_id) { this->agent_id_ = agent_id; }
+  //! Get id
+  graph::vertex_t get_id() const { return agent_id_; }
+
+  //! Set origin and initialize current node to origin
+  //! \param[in] origin Origin vertex assigned to agent
+  void set_origin(graph::vertex_t origin) { 
```

Comment:
Pass this in the constructor as a const &

Reply:
What should be passed in the constructor? Origin (const &), destination (const &), departure time (double value) and ID (const &)?

Reply:
This is not the constructor, this is just a function.

Reply:
We can have one function to set origin and destinations